third_party/amd/lib/TritonAMDGPUTransforms/AccelerateAMDMatmul.cpp

```diff
@@ -867,6 +817,134 @@ class BlockedToWMMA : public OpRewritePattern<tt::DotOp> {
     return success();
   }
 };
+
+class AccelerateBlocked : public OpRewritePattern<DotOp> {
+  StringRef arch;
+
+public:
+  AccelerateBlocked(MLIRContext *context, StringRef arch,
+                    PatternBenefit benefit = 1)
+      : OpRewritePattern(context, benefit), arch(arch) {}
+
+  bool isFloat(Type t) const { return t.isIntOrFloat() && !t.isIntOrIndex(); }
+
+  Value castToElTy(PatternRewriter &rewriter, Value v, Type elTy) const {
+    Location loc = v.getLoc();
+    auto srcTy = cast<RankedTensorType>(v.getType());
+    auto dstTy = srcTy.cloneWith(std::nullopt, elTy);
+    if (srcTy == dstTy)
+      return v;
+    auto srcElTy = srcTy.getElementType();
+    auto dstElTy = dstTy.getElementType();
+    if (isFloat(srcElTy) && isFloat(dstElTy))
+      return rewriter.create<FpToFpOp>(loc, dstTy, v);
+    if (!isFloat(srcElTy) && isFloat(dstElTy))
+      return rewriter.create<arith::SIToFPOp>(loc, dstTy, v);
+    if (isFloat(srcElTy) && !isFloat(dstElTy))
+      return rewriter.create<arith::FPToSIOp>(loc, dstTy, v);
+    assert(false && "int -> int cast is unexpected in FMA legalization");
+    return Value();
+  }
+
+  bool isLegalFMAForm(DotOp dotOp) const {
+    auto expectedElTy = dotOp.getA().getType().getElementType();
+    for (auto operand : dotOp.getOperands()) {
+      auto opTy = cast<RankedTensorType>(operand.getType());
+      auto elTy = opTy.getElementType();
+      if (elTy != expectedElTy)
+        return false;
+      if (!elTy.isF16() && !elTy.isF32())
+        return false;
+    }
+    return true;
+  }
+
+  LogicalResult matchAndRewrite(DotOp dotOp,
+                                PatternRewriter &rewriter) const override {
+    auto a = dotOp.getA();
+    auto b = dotOp.getB();
+    auto c = dotOp.getC();
+    auto d = dotOp.getD();
+
+    if (!isa<BlockedEncodingAttr>(d.getType().getEncoding()))
+      return failure();
+
+    Type aElTy = a.getType().getElementType();
+    Type bElTy = b.getType().getElementType();
+    Type cElTy = c.getType().getElementType();
+    Type dElTy = d.getType().getElementType();
+
+    int rank = a.getType().getShape().size();
+    int k = a.getType().getShape()[rank - 1];
+
+    bool dotAvailable = arch == "gfx908" || arch == "gfx90a" ||
```

Comment:
Use `deduceISAFamily` in `TargetUtils.h` instead of manually checking the string.